### DISCUSSION TITLE
Error when trying to open CMIP6-GLADE datastore

### DISCUSSION THREAD

**Discussion by PaulScemama** [intake/intake-esm#590] [Q&A] [answered]
Hi! 

I'm trying to access the CMIP6-GLADE datastore via the url in the FAQ. However, when I run

```python
cat = intake.open_esm_datasource("/glade/collections/cmip/catalog/intake-esm-datastore/catalogs/glade-cmip6.json")
```

I get 

```
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)


ValidationError: 5 validation errors for Init
records
  field required (type=value_error.missing)
variable_column_name
  field required (type=value_error.missing)
path_column_name
  field required (type=value_error.missing)
data_format
  field required (type=value_error.missing)
format_column_name
  field required (type=value_error.missing)
```

Any idea what I'm doing wrong?

**Comment by andersy005** [comment, marked as answer]
@PaulScemama, 

use `open_esm_datastore()` instead of `open_esm_datasource()`

```python
cat = intake.open_esm_datastore("/glade/collections/cmip/catalog/intake-esm-datastore/catalogs/glade-cmip6.json")
```

**Reply by PaulScemama** [reply]
Ah yes I missed that thank you. Now I'm getting

```
FileNotFoundError: [Errno 2] No such file or directory: '[/glade/collections/cmip/catalog/intake-esm-datastore/catalogs/glade-cmip6.json]
```

Do I need to download the `.json`? If so, where do I find it? Thanks!

**Comment by andersy005** [comment]
you bet! 

> Do I need to download the .json? If so, where do I find it? 

are you running this from your local computer? if so, this will not work. That file is only accessible if you have access to [NCAR's GLADE filesystem](https://arc.ucar.edu/knowledge_base/68878466).

**Reply by PaulScemama** [reply]
Yes I am on my local computer. That makes sense! Do you have any idea what the best way would be for me to access CMIP6 files with this library? I'm new to climate data. Is there a centralized place where I can request data access or pull CMIP6 data from? No worries if you don't have the answers :). Thanks again!

**Reply by andersy005** [reply]
the data are available in both google cloud and aws. so, you should be able to access them from your local computer without needing to download them. 


```python
# for Google Cloud:
cat = intake.open_esm_datastore("https://storage.googleapis.com/cmip6/pangeo-cmip6.json")
```

```python
# for AWS S3:
cat = intake.open_esm_datastore("https://cmip6-pds.s3.amazonaws.com/pangeo-cmip6.json")
```

for more information, check this website https://pangeo-data.github.io/pangeo-cmip6-cloud/index.html and repository: https://github.com/pangeo-data/pangeo-cmip6-cloud
